analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs

```diff
@@ -55,9 +51,7 @@ private static void DetectTypeChanges(Dictionary<string, RuleDescriptor> rules,
             var newRules = items.Where(x => x.ExpectedType is null);
             newRules.Should().BeEmpty($"you need to add the rules in {expectedTypesName}");
 
-            items.Should().NotContain(
-                x => x.ActualType == null && !DeletedRules.Contains($"S{x.RuleId}"),
-                "YOU SHOULD NEVER DELETE RULES! Rules should not be deleted without careful consideration and prior deprecation. We need to notify the product teams (SQ, SC, SL) as well.");
+            items.Should().NotContain(x => x.ActualType == null);
```

Comment:
It's hard to understand what this means. To keep it consistent, extract it to a variable and assert it should be empty, with the string "because" explanation.